app/Http/Controllers/AdminController.php

```diff
@@ -1426,16 +1426,15 @@ public function monitor(): View
     {
         $user = Auth::user();
         if ($user->admin) {
-            $content = $this->monitor_currently_processing_submissions();
-            $content .= $this->monitor_pending_submissions();
-            $content .= $this->monitor_average_wait_times();
-            $content .= $this->monitor_submissionprocessor_table();
-            $content .= $this->monitor_submission_table();
-            return view('cdash', [
-                'xsl' => true,
-                'xsl_content' => $content,
-                'title' => 'System Monitor'
-            ]);
+            if (config('queue.default') === 'database') {
+                return view('admin.monitor');
+            } else {
+                return view('cdash', [
+                        'xsl' => true,
+                        'xsl_content' => 'System Monitor only available when QUEUE_CONNECTION=database',
+                        'title' => 'System Monitor'
+                ]);
+            }
```

Comment:
It would be better to do this kind of logic in the Blade template itself--conditional rendering like this is exactly what Blade is useful for.